File: game_server/ServerProtocol/src/Common/MarsObj/Mars_HomeBuilding.java
package Common.MarsObj;

import java.nio.ByteBuffer;
/*********
 * 火星建筑-主基地
 **/
public class Mars_HomeBuilding implements ALBasicProtocolPack._IALProtocolStructure {
/** 建筑ID */
private long buildingId;
/** 普通功率开启 */
private boolean isNormalOn;
/** 最高功率开启 */
private boolean isOverdriveOn;
/** 最后一次收集资源的时间（秒） */
private int lastCollectTimeS;


public Mars_HomeBuilding() {
	buildingId = (long)0;
	isNormalOn = false;
	isOverdriveOn = false;
	lastCollectTimeS = 0;
}

public Mars_HomeBuilding(
	 long _buildingId
	, boolean _isNormalOn
	, boolean _isOverdriveOn
	, int _lastCollectTimeS
) {	buildingId = _buildingId;
	isNormalOn = _isNormalOn;
	isOverdriveOn = _isOverdriveOn;
	lastCollectTimeS = _lastCollectTimeS;
}

public final byte getMainOrder() { return (byte)0; }

public final byte getSubOrder() { return (byte)0; }

/** 建筑ID */
public long getBuildingId() { return buildingId; }
/** 建筑ID */
public void setBuildingId(long _buildingId) { buildingId = _buildingId; }
/** 普通功率开启 */
public boolean getIsNormalOn() { return isNormalOn; }
/** 普通功率开启 */
public void setIsNormalOn(boolean _isNormalOn) { isNormalOn = _isNormalOn; }
/** 最高功率开启 */
public boolean getIsOverdriveOn() { return isOverdriveOn; }
/** 最高功率开启 */
public void setIsOverdriveOn(boolean _isOverdriveOn) { isOverdriveOn = _isOverdriveOn; }
/** 最后一次收集资源的时间（秒） */
public int getLastCollectTimeS() { return lastCollectTimeS; }
/** 最后一次收集资源的时间（秒） */
public void setLastCollectTimeS(int _lastCollectTimeS) { lastCollectTimeS = _lastCollectTimeS; }


public final int GetBufSize() {
	int _size = 14;

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 16;

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) buildingId = _buf.getLong();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) isNormalOn = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) isOverdriveOn = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) lastCollectTimeS = _buf.getInt();
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putLong(buildingId);
	_buf.put(isNormalOn?(byte)1:(byte)0);
	_buf.put(isOverdriveOn?(byte)1:(byte)0);
	_buf.putInt(lastCollectTimeS);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)0);
	_buf.put((byte)0);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)0);
	_recBuf.put((byte)0);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_server/ClientProtocol/Common/MarsObj/Mars_HomeBuilding.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace Common.MarsObj
{

/// <summary>
/// 火星建筑-主基地
/// </summary>
public class Mars_HomeBuilding : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 建筑ID
/// </summary>
private long buildingId;
/// <summary>
/// 普通功率开启
/// </summary>
private bool isNormalOn;
/// <summary>
/// 最高功率开启
/// </summary>
private bool isOverdriveOn;
/// <summary>
/// 最后一次收集资源的时间（秒）
/// </summary>
private int lastCollectTimeS;


public Mars_HomeBuilding() {
	buildingId = (long)0;
	isNormalOn = false;
	isOverdriveOn = false;
	lastCollectTimeS = 0;
}

public Mars_HomeBuilding(
	long _buildingId
	, bool _isNormalOn
	, bool _isOverdriveOn
	, int _lastCollectTimeS
) {	buildingId = _buildingId;
	isNormalOn = _isNormalOn;
	isOverdriveOn = _isOverdriveOn;
	lastCollectTimeS = _lastCollectTimeS;
}

public byte getMainOrder() { return (byte)0; }

public byte getSubOrder() { return (byte)0; }

/// <summary>
/// 建筑ID
/// </summary>
public long getBuildingId() { return buildingId; }
/// <summary>
/// 建筑ID
/// </summary>
public void setBuildingId(long _buildingId) { buildingId = _buildingId; }
/// <summary>
/// 普通功率开启
/// </summary>
public bool getIsNormalOn() { return isNormalOn; }
/// <summary>
/// 普通功率开启
/// </summary>
public void setIsNormalOn(bool _isNormalOn) { isNormalOn = _isNormalOn; }
/// <summary>
/// 最高功率开启
/// </summary>
public bool getIsOverdriveOn() { return isOverdriveOn; }
/// <summary>
/// 最高功率开启
/// </summary>
public void setIsOverdriveOn(bool _isOverdriveOn) { isOverdriveOn = _isOverdriveOn; }
/// <summary>
/// 最后一次收集资源的时间（秒）
/// </summary>
public int getLastCollectTimeS() { return lastCollectTimeS; }
/// <summary>
/// 最后一次收集资源的时间（秒）
/// </summary>
public void setLastCollectTimeS(int _lastCollectTimeS) { lastCollectTimeS = _lastCollectTimeS; }


public int GetBufSize() {
	int _size = 14;

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 16;

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	buildingId = _buf.getLong();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	isNormalOn = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	isOverdriveOn = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	lastCollectTimeS = _buf.getInt();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putLong(buildingId);
	_buf.put(isNormalOn?(byte)1:(byte)0);
	_buf.put(isOverdriveOn?(byte)1:(byte)0);
	_buf.putInt(lastCollectTimeS);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)0);
	_buf.put((byte)0);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)0);
	_recBuf.put((byte)0);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("buildingId").Append(":").Append(buildingId.ToString()).Append(", ");
	builder.Append("isNormalOn").Append(":").Append(isNormalOn.ToString()).Append(", ");
	builder.Append("isOverdriveOn").Append(":").Append(isOverdriveOn.ToString()).Append(", ");
	builder.Append("lastCollectTimeS").Append(":").Append(lastCollectTimeS.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

